services/gossip/codec_test.go

```diff
@@ -0,0 +1,105 @@
+package gossip
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/orbs-network/orbs-network-go/test"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"testing"
+)
+
+var blockPairTable = []struct {
```

Comment:
really, really hard to figure out what this means
Please create a factory method, or a builder-of-builders, so that a reader can understand from a glance at the table what is expected. 
In addition, consider adding an explanation (string) to the errors (both encode and decode) so that the reader can understand *why* we expect any specific entry in the table to fail

Reply:
yes.. I thought 10 times before adding this unit test.. I'm not in the gossip team but I needed to encode and decode a full block in gossip for my consensus implementation and I just couldn't leave this without a unit test

I'm wondering if we should leave this to them to improve the way the codec is written.. or write it according to my own liking..

Reply:
I'll refactor the test and add builders, hold on